File: dohoudaniel/C/perform_calc.c
#include <stdio.h>   /* To include standard input and output functions */
#include <stdlib.h>  /* To include standard library functions */
#include "math_functions.h"    /* Include function definition file */
/**
 * performCalculation - Performs arithmetic calculation based on user input
 * @num1: First number
 * @num2: Second number
 * @operator: Arithmetic operator
 *
 * Return: Result of arithmetic calculation
 */
double performCalculation(double num1, double num2, char operator)
{
	double solution = 0.0;

	/* Using a switch statement to perform the arithmetic operations */
	switch (operator)
	{
		case '+':
			solution = add(num1, num2);
			break;
		case '-':
			solution = subtract(num1, num2);
			break;
		case '*':
			solution = mult(num1, num2);
			break;
		case '/':
			solution = divide(num1, num2);
			break;
		default:
			printf("You have entered an invalid operator. Try again.\n");
			exit(EXIT_FAILURE);    /* Program exits on failure. */
	}

	return (solution);
}

File: dohoudaniel/Python/perform_calc.py
#!/usr/bin/python3
"""
Module: perform_calc
This module performs the arithmetic operations
"""

from function_classes import *
from time import sleep


def perform_calc(num1, num2, operator):
    """
    Performs the arithmetic calculations
    """
    if operator == '+':
        return MathOperations.add(num1, num2)
    elif operator == '-':
        return MathOperations.subtract(num1, num2)
    elif operator == '*':
        return MathOperations.mult(num1, num2)
    elif operator == '/':
        return MathOperations.divide(num1, num2)
    else:
        print("Invalid arithmetic operation!!")
        sleep(2)
        print("Try again!")
        sleep(2)
        return None
        # exit()
        # pass
